### PR TITLE
feature(nyp): add apple key to door treasure and polish

### PR DESCRIPTION
**Pull request by Will-Nie**
## Description

This pr add the apple door to key treasure env, currently we support 

Add three MiniGrid envs, one base and two specific apple key to door treasure:

MiniGrid-AKTDT-13x13-v0: 13 by 13 minigrid
MiniGrid-AKTDT-19x19-v0: 19 by 19 minigrid

WeChatWorkScreenshot_7f4762d0-4f7d-4593-afd5-cc620c9ac0e7
WeChatWorkScreenshot_282d2a92-974c-46ee-9108-3564c9393cfa
WeChatWorkScreenshot_a1013544-1fd8-435b-99dd-0e72fc0affbc

but it is very easy to make a modification

See below for the future ToDo

---update on 25 Nov--
1 . modify reward from gym_minigrid.MiniGridEnv 
2 . solve the position issue of the ball 
3 . make no of apples a hyperparameter 
4.   writing test 
5. pick up the object -- redesign
6. dig into seed issue
WeChatWorkScreenshot_175eaefd-1fb3-41bc-b0f9-053e75292462
WeChatWorkScreenshot_664effeb-b547-45cd-bac8-eb008a7ef09c
WeChatWorkScreenshot_08aac901-9f96-4d99-9d49-0bb58251988f
WeChatWorkScreenshot_162ae399-5c6f-41cc-bf7b-7a2aa3ca6498
WeChatWorkScreenshot_dcc219a3-146b-4928-b9d8-a6bf0c124208

## TODO

1. optimisation: world vector
2. object encoder observation dim research

## Check List

- [ x ] merge the latest version source branch/repo, and resolve all the conflicts
- [ x ] pass style check
- [ x ] pass all the tests

### CODE REVIEW

File: dizoo/minigrid/envs/minigrid_env.py [hunk 1]
```diff
@@ -208,6 +277,11 @@ def __init__(self, cfg: dict) -> None:
     def reset(self) -> np.ndarray:
         if not self._init_flag:
             self._env = gym.make(self._env_id)
+            if self._env_id == 'MiniGrid-AKTDT-13x13-v0':
+                self._env = ViewSizeWrapper(
+                    self._env, agent_view_size=5
+                )  # customize the agent field of view size, note this must be an odd number # This also related to the observation space, see gym_minigrid.wrappers for more details
+            self._env.render()
```
Comment (Will-Nie) [inline dizoo/minigrid/envs/minigrid_env.py:284]: will delete this render before merge

File: dizoo/minigrid/envs/app_key_to_door_treasure.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        super().__init__(agent_pos=(2, 8), goal_pos=(7, 1), grid_size=13)
+
+
+class AppleKeyToDoorTreasure_19x19(AppleKeyToDoorTreasure):
+
+    def __init__(self):
+        super().__init__(agent_pos=(2, 14), goal_pos=(10, 1), grid_size=19)
+
+
+#AppleKeyToDoorTreasure()._gen_grid(13,13)  #Note that Minigrid has set seeds automatically
```
Comment (PaParaZz1) [inline dizoo/minigrid/envs/app_key_to_door_treasure.py:116]: remove unused commented code

File: dizoo/setup.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from setuptools import setup
```
Comment (PaParaZz1) [inline dizoo/setup.py:1]: remove this setup.py, you have not installed any packages and the requirements is already indicated in `DI-engine/setup.py`

**Review by PaParaZz1 — APPROVED** [review]

File: dizoo/minigrid/envs/app_key_to_door_treasure.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from gym_minigrid.minigrid import *
+from gym_minigrid.minigrid import WorldObj
+'''
+class Apple(WorldObj):
```
Comment (Will-Nie) [inline dizoo/minigrid/envs/app_key_to_door_treasure.py:7]: will delete

**Pull request merged**